src/libAtomVM/CMakeLists.txt

```diff
@@ -140,23 +155,16 @@ else()
     set(ATOMVM_VERSION ${ATOMVM_BASE_VERSION})
 endif()
 
+target_include_directories(libAtomVM PUBLIC ${CMAKE_CURRENT_BINARY_DIR})
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/version.h.in ${CMAKE_CURRENT_BINARY_DIR}/version.h)
-include_directories(${CMAKE_CURRENT_BINARY_DIR})
 
 if (NOT ${CMAKE_SYSTEM_NAME} STREQUAL "Generic")
-    set(
-        ATOMVM_PLATFORM_LIB
-        libAtomVM${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR}
-    )
-else()
-    set(ATOMVM_PLATFORM_LIB "")
+    target_link_libraries(libAtomVM PUBLIC libAtomVM${CMAKE_SYSTEM_NAME}-${CMAKE_SYSTEM_PROCESSOR})
 endif()
 
-add_library(libAtomVM ${SOURCE_FILES} ${HEADER_FILES})
```

Comment:
Why is this removed?  Doesn't that result in not creating a target for `libAtomVM` in the generated make files (or Xcode project)?

Reply:
I moved this at the top (line 89) to be able to refer to target libAtomVM in target_* commands.